repository/repository.go

```diff
@@ -17,31 +18,72 @@ type Repository interface {
 	GetLoadBalancer(id string) (LoadBalancer, error)
 }
 
+var (
+	ErrNoFieldsToUpdate               = errors.New("no fields to update")
+	ErrInvalidAppStatus               = errors.New("invalid app status")
+	ErrInvalidPayPlanType             = errors.New("invalid pay plan type")
+	ErrNotEnterprisePlan              = errors.New("custom limits may only be set on enterprise plans")
+	ErrEnterprisePlanNeedsCustomLimit = errors.New("enterprise plans must have a custom limit set")
+)
+
 // TODO: identify fields that should be stored encrypted in-memory
 type Application struct {
-	ID                   string               `json:"id"`
-	UserID               string               `json:"userID"`
-	Name                 string               `json:"name"`
-	ContactEmail         string               `json:"contactEmail"`
-	Description          string               `json:"description"`
-	Owner                string               `json:"owner"`
-	URL                  string               `json:"url"`
-	Status               AppStatus            `json:"status"`
-	Dummy                bool                 `json:"dummy"`
-	PayPlanType          PayPlanType          `json:"payPlanType,omitempty"`
-	FirstDateSurpassed   time.Time            `json:"firstDateSurpassed"`
+	ID                 string    `json:"id"`
+	UserID             string    `json:"userID"`
+	Name               string    `json:"name"`
+	ContactEmail       string    `json:"contactEmail"`
+	Description        string    `json:"description"`
+	Owner              string    `json:"owner"`
+	URL                string    `json:"url"`
+	Dummy              bool      `json:"dummy"`
+	Status             AppStatus `json:"status"`
+	FirstDateSurpassed time.Time `json:"firstDateSurpassed"`
+	CreatedAt          time.Time `json:"createdAt"`
+	UpdatedAt          time.Time `json:"updatedAt"`
+
 	GatewayAAT           GatewayAAT           `json:"gatewayAAT"`
 	GatewaySettings      GatewaySettings      `json:"gatewaySettings"`
+	Limit                AppLimit             `json:"limit"`
 	NotificationSettings NotificationSettings `json:"notificationSettings"`
-	Limits               AppLimits            `json:"limits"`
-	CreatedAt            time.Time            `json:"createdAt"`
-	UpdatedAt            time.Time            `json:"updatedAt"`
 }
 
 func (a *Application) Table() Table {
 	return TableApplications
 }
 
+func (a *Application) DailyLimit() int {
+	if a.Limit.PayPlan.Type == Enterprise {
+		return a.Limit.CustomLimit
+	}
+
+	return a.Limit.PayPlan.Limit
+}
+
+func (a *Application) IsInvalid() error {
```

Comment:
Not a blocker, but it is a bit non-conventional to use such a name. `Validate` may be a better choice.